gnd/src/androidTest/java/com/google/android/gnd/AddFeatureTest.java

```diff
@@ -0,0 +1,152 @@
+
+  // Sets the preferences so no login is required and an active project is selected.
+  public @Rule(order = 1)
+  SetPreferencesRule preferencesRule = new SetPreferencesRule();
+
+  public @Rule(order = 2)
+  ActivityScenarioRule scenarioRule =
+    new ActivityScenarioRule(MainActivity.class);
+
+  @Module
+  @InstallIn(ApplicationComponent.class)
+  abstract class TestDependenciesModule {
+
+    @Binds
+    @Singleton
+    abstract AuthenticationManager bindAuthenticationManager(
+      FakeAuthenticationManager authenticationManager);
+
+    @Binds
+    @Singleton
+    abstract RemoteDataStore bindRemoteDataStore(FakeRemoteDataStore remoteDataStore);
+
+    @Binds
+    @Singleton
+    abstract RemoteStorageManager bindRemoteStorageManager(
+      FakeRemoteStorageManager remoteStorageManager);
+
+    @Binds
+    @Singleton
+    abstract OfflineUuidGenerator offlineUuidGenerator(FakeUuidGenerator uuidGenerator);
+  }
+
+  @InstallIn(ApplicationComponent.class)
+  @Module
+  public class LocalDatabaseModule {
+
+    @Provides
+    @Singleton
+    LocalDatabase localDatabase(@ApplicationContext Context context) {
+      return Room.inMemoryDatabaseBuilder(context, LocalDatabase.class).build();
+    }
+  }
+
+  // Given: a logged in user - with an active project with no map markers.
+  // When: they tap on the centre of the map.
+  // Then: nothing happens - the feature fragment is not displayed.
+  @Test
+  public void tappingCrosshairOnEmptyMapDoesNothing() throws InterruptedException {
+    ActivityScenario<MainActivity> scenario = scenarioRule.getScenario();
+
+    // Tap on the crosshair at the centre of the map.
+    onView(withId(R.id.map_crosshairs)).perform(click());
+
+    // Verify that the title is not displayed.
+    onView(withId(R.id.feature_title)).check(matches(not(isCompletelyDisplayed())));
+  }
+
+  // Given: a logged in user - with an active project which doesn't direct the user to add an
+  // observation when adding a feature.
+  // When: they tap the "Add feature" FAB and choose a feature type.
+  // Then: the observation marker is displayed on the map screen.
+  @Test
+  public void addFeatureWithNoFields() throws InterruptedException {
+    ActivityScenario<MainActivity> scenario = scenarioRule.getScenario();
+
+    // Tap on the "Add feature" button.
+    onView(withId(R.id.add_feature_btn)).perform(click());
+
+    // Tap on the layer type.
+    onData(allOf(is(instanceOf(String.class)), is(FakeData.LAYER_NO_FIELDS_NAME))).perform(click());
+
+    // Tap on the crosshair at the centre of the map.
+    onView(withId(R.id.map_crosshairs)).perform(click());
+
+    // TODO: figure out how to remove this
+    //  It's probably because the feature fragment animation does not block the test and the
+    //  feature title isn't updated until the animation has completed (and it isn't disabled by the
+    //  normal `transition_animation_scale 0` etc commands).
+    Thread.sleep(100);
```

Comment:
Have you tried idling resources? Given you're dealing with maps and fragments it might be tricky

Reply:
I think your idea for an `AnimationManager` which switches off animations for testing is probably the best one.

Reply:
In iosched animation is disabled manually for the bottom sheet: https://github.com/google/iosched/blob/89df01ebc19d9a46495baac4690c2ebfa74946dc/mobile/src/main/java/com/google/samples/apps/iosched/widget/BottomSheetBehavior.kt#L209 (thanks @JoseAlcerreca)

Reply:
I tried taking the current version of [BottomSheetBehavior](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/bottomsheet/BottomSheetBehavior.java#L1312) and modifying it so that animations are disabled. This resulted in the test passing without `Thread.sleep` on my Pixel 4A but still failing on the emulator (tried API 23 and 29). I feel like there's a background thread being spun up somewhere after tapping on the map pin which delays the display of the bottom sheet just long enough for the test to fail.